pymongo/collection.py

```diff
@@ -2360,6 +2361,189 @@ def index_information(
             info[index.pop("name")] = index
         return info
 
+    def list_search_indexes(
+        self,
+        name: Optional[str] = None,
+        session: Optional[ClientSession] = None,
+        comment: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> CommandCursor[Mapping[str, Any]]:
+        """Return a cursor over search indexes for the current collection.
+
+        :Parameters:
+          - `name` (optional): If given, the name of the index to search
+            for.  Only indexes with matching index names will be returned.
+            If not given, all search indexes for the current collection
+            will be returned.
+          - `session` (optional): a :class:`~pymongo.client_session.ClientSession`.
+          - `comment` (optional): A user-provided comment to attach to this
+            command.
+
+        :Returns:
+          A :class:`~pymongo.command_cursor.CommandCursor` over the result
+          set.
+
+        .. note:: requires a MongoDB server version 7.0+ Atlas cluster.
+
+        .. versionadded:: 4.5
+        """
+        if name is None:
+            pipeline: _Pipeline = [{"$listSearchIndexes": {}}]
+        else:
+            pipeline = [{"$listSearchIndexes": {"name": name}}]
+        kwargs["codec_options"] = CodecOptions(SON)
```

Comment:
Instead of adding a new codec_options feature to aggregate, let's use `self.with_options(codec_options=...)` like list_indexes does. Also I suggest using DEFAULT_CODEC_OPTIONS instead of SON since SON is redundant now.

Reply:
Done